vpnglean/CMakeLists.txt

```diff
@@ -6,37 +6,37 @@ add_library(vpnglean STATIC IMPORTED GLOBAL)
 
 set_target_properties(vpnglean PROPERTIES FOLDER "Libs")
 
-if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
-    set(OPENSSL_USE_STATIC_LIBS TRUE)
-    find_package(OpenSSL REQUIRED)
-    get_filename_component(OPENSSL_LIB_DIR ${OPENSSL_SSL_LIBRARY} DIRECTORY)
-endif()
-
-if (CMAKE_BUILD_TYPE STREQUAL "Debug")
-    set(CARGO_CMD cargo build)
-    set(TARGET_DIR "debug")
-else ()
-    set(CARGO_CMD cargo build --release)
-    set(TARGET_DIR "release")
-endif ()
-
-set (HEADER_FILE vpnglean.h)
+set(HEADER_FILE vpnglean.h)
 set(LIBNAME ${CMAKE_STATIC_LIBRARY_PREFIX}vpnglean${CMAKE_STATIC_LIBRARY_SUFFIX})
-get_filename_component(GENERATED_DIR ${CMAKE_CURRENT_BINARY_DIR}/${TARGET_DIR} ABSOLUTE)
 
-set(VPNGLEAN_ENV BUILD_ID=${BUILD_ID} APP_VERSION=${CMAKE_PROJECT_VERSION} CARGO_TARGET_DIR=${CMAKE_CURRENT_BINARY_DIR})
-set(OPENSSL_ENV OPENSSL_STATIC=yes OPENSSL_LIB_DIR=${OPENSSL_LIB_DIR} OPENSSL_INCLUDE_DIR=${OPENSSL_INCLUDE_DIR})
 add_custom_target(vpnglean_ffi
-    BYPRODUCTS ${GENERATED_DIR}/${LIBNAME} ${CMAKE_CURRENT_BINARY_DIR}/${HEADER_FILE}
-    COMMAND ${CMAKE_COMMAND} -E env ${VPNGLEAN_ENV} ${OPENSSL_ENV} ${CARGO_CMD}
+    BYPRODUCTS
+        ${CMAKE_CURRENT_BINARY_DIR}/debug/${LIBNAME}
+        ${CMAKE_CURRENT_BINARY_DIR}/release/${LIBNAME}
+        ${CMAKE_CURRENT_BINARY_DIR}/${HEADER_FILE}
+    COMMAND ${CMAKE_COMMAND} -E env
+                BUILD_ID=${BUILD_ID}
+                APP_VERSION=${CMAKE_PROJECT_VERSION}
+                CARGO_TARGET_DIR=${CMAKE_CURRENT_BINARY_DIR}
+            cargo build $<IF:$<CONFIG:Debug>,,--release>
```

Comment:
FWIW I copied that if/else from signature.cmake. We might want to clean up there as well.

Reply:
Good catch!

Reply:
I think I'm going to leave that cleanup for a future PR for now.

Reply:
Np!